Build failure: wezterm

### Steps To Reproduce

Steps to reproduce the behavior:
1. Build `wezterm` for `x86_64-darwin`, on MacOS Ventura (13.3.x)

### Build log

https://gist.github.com/remi-gelinas/fd07dce3a836a7c87f41b1a062350d9e

### Additional context

Tested builds on both a `x86_64-darwin` `hostSystem` and an `aarch64-darwin` `hostSystem` and got the same error. Metadata was obtained on an `aarch64-darwin` machine compiling for both architectures, as x86 Darwin can be compiled for on Apple Silicon.

### Notify maintainers

@SuperSandro2000 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.4.0, macOS 13.3.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.3`
 - channels(user): `""`
 - nixpkgs: `/nix/store/00d5ysz5ws1xpfv229m9ml6in7qlkz1b-source`
```

No idea and I am not really motivated to dig into why this is broken myself. If someone figures it out, please fill a PR and I'll take a look at it.
```
  = note: Undefined symbols for architecture x86_64:
            "___darwin_check_fd_set_overflow", referenced from:
                _ssh_poll in liblibssh_rs_sys-fa10a07daad3af81.rlib(poll.o)
          ld: symbol(s) not found for architecture x86_64
          clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/pkgs/applications/terminal-emulators/wezterm/default.nix
+++ b/pkgs/applications/terminal-emulators/wezterm/default.nix
@@ -22,6 +22,7 @@
 , CoreGraphics
 , Cocoa
 , Foundation
+, System
 , libiconv
 , UserNotifications
 , nixosTests
@@ -81,10 +82,13 @@ rustPlatform.buildRustPackage rec {
     CoreGraphics
     Foundation
     libiconv
+    System
     UserNotifications
   ];
 
   buildFeatures = [ "distro-defaults" ];
+
+  env.NIX_LDFLAGS = lib.optionalString stdenv.isDarwin "-framework System";
 
   postInstall = ''
     mkdir -p $out/nix-support

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2708,7 +2708,7 @@ with pkgs;
   wayst = callPackage ../applications/terminal-emulators/wayst { };
 
   wezterm = darwin.apple_sdk_11_0.callPackage ../applications/terminal-emulators/wezterm {
-    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa CoreGraphics Foundation UserNotifications;
+    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa CoreGraphics Foundation UserNotifications System;
   };
 
   x3270 = callPackage ../applications/terminal-emulators/x3270 { };

